uber/scheduler.py

```diff
@@ -1,21 +1,40 @@
-from sideboard.lib import DaemonTask, on_startup
+from sideboard.lib import DaemonTask, on_startup, log
 from uber.decorators import timed, swallow_exceptions
 from datetime import datetime
 import time
 import schedule
 from collections import defaultdict
+from functools import wraps
+import threading
 
 # don't start the email sending in the main thread. everything else is cool.
 # NOTE: only applies to cherrypy process ONLY, doesn't apply when started via 'sep' or other processes.
 # uncomment this if you want to do email sending in its own process
-# _default_cherrypy_exclude_categories = ["automated_email_sending"]
-_default_cherrypy_exclude_categories = None
+_default_cherrypy_exclude_categories = ["automated_email_sending"]
+
+
+def run_threaded(func, thread_name):
+    """
+    Decorate a function to run asynchronously in a new thread and return immediately
+    NOTE: Caller is responsible for all thread safety, this is very lightweight.
+    """
+    @wraps(func)
+    def with_threading(*args, **kwargs):
+        job_thread = threading.Thread(target=func, *args, **kwargs)
+        if thread_name:
+            job_thread.name = '{}'.format(thread_name)
+        job_thread.start()
+        log.debug('Started background thread for task: {}'.format(thread_name))
+    return with_threading
 
 
 class UberSchedulerJob(schedule.Job):
-    def do(self, job_func, *args, **kwargs):
-        # our scheduler needs to not let individual tasks throw exceptions, so wrap it here
-        wrapped_fn = timed(swallow_exceptions(job_func))
+    def do(self, job_func, run_in_own_thread=False, thread_name=None, *args, **kwargs):
+        # our scheduler needs to not let individual tasks throw exceptions, so wrap it here, and time it
+        wrapped_fn = timed(swallow_exceptions(job_func), prepend_txt='TASK DONE')
+
+        if run_in_own_thread:
```

Comment:
As a naming suggestion, I recommend changing this to ``run_in_new_thread`` because I originally misinterpreted "run in own thread" to mean "run in the owner's thread and NOT in a new thread".  Which is clearly wrong and kind of a dumb interpretation on my part, since presumably if you meant that you would have said ``run_in_caller_thread`` or something, but I was still confused.